Assets/LeapMotion/Core/Scripts/DataStructures/Pool.cs

```diff
@@ -85,7 +83,29 @@ public static class Pool<T> where T : new() {
       return value;
     }
 
+    private static T spawnValue() {
+      T value;
+      int count = 0;
+      do {
+        if (_pool.Count > 0) {
+          value = _pool.Pop();
+        }
+        else {
+          value = new T();
+        }
+        count++;
+      } while (value == null && count < 100);
+      if (count == 100) {
```

Comment:
I ran into an issue where, because I had mishandled pooled objects, I had accidentally added null object to a Pool for those objects. The idea was to make sure we _really_ construct a new object and lazily clean up the pool if it contains null objects, and the counter prevents Unity from hard-locking if we somehow are unable to construct a non-null object (which could be strictly impossible?)